client/app/controllers/projects.js

```diff
@@ -1,41 +1,80 @@
+  FINAL_APPROVAL: 'Final Approval',
+  REVIEWED_NO_REVISIONS_REQUIRED: 'Reviewed - No Revisions Required',
+  REVIEWED_REVISIONS_REQUIRED: 'Reviewed - Revisions Required',
+  WITHDRAWN: 'Withdrawn',
+  SUBMITTED: 'Submitted',
+  UNDER_REVIEW: 'Under Review',
+  MISTAKE: 'Mistake',
+};
+
+const PACKAGE_VISIBILITY_CODES = {
+  APPLICANT_ONLY: 717170002,
+  GENERAL_PUBLIC: 717170003,
+};
 
 export default class ProjectsController extends Controller {
   queryParams = ['email'];
 
   // TODO: organize this business logic as computed properties on the projects model
   // projects for applicant to do
   get applicantProjects () {
-    return this.model.filter((project) => project.packages.some((projectpackage) => {
-      if (projectpackage.statuscode === 'Saved' || projectpackage.statuscode === 'Package Preparation') {
+    return this.model.filter((project) => project.pasPackages.some((projectPackage) => {
+      if (
+        projectPackage.statuscode === PACKAGE_STATUS_CODES.PACKAGE_PREPARATION
+        && [
+          PACKAGE_VISIBILITY_CODES.APPLICANT_ONLY,
+          PACKAGE_VISIBILITY_CODES.GENERAL_PUBLIC,
+        ].includes(projectPackage['dcp-visibility'])
+      ) {
         return true;
-      } return false;
+      }
+      return false;
     }));
   }
 
-  // projects for nyc planning to do
+  // projects for nyc planning to do.
+  // These are all other returned projects that are not in applicantProjects.
+  // (I.e. this list includes projects with no packages).
   get planningProjects () {
-    return this.model.filter((project) => project.packages.some((projectpackage) => {
-      if (projectpackage.statuscode === 'Submitted' || projectpackage.statuscode === 'Under Review' || projectpackage.statuscode === 'Revision Required') {
-        return true;
-      } return false;
-    }));
-  }
-
-  // projects that have no pas packages are shown as "upcoming"
-  get noPasProjects () {
     return this.model.filter((project) => {
-      // filter to get only pas packages
-      const pasPackages = project.packages.filter((projectpackage) => {
-        if (projectpackage['dcp-packagetype'] === 'PAS Package') {
-          return true;
-        } return false;
-      });
-
-      // return the project if there are no pasPackages
-      if (pasPackages.length === 0) {
+      // if pasPackages is empty, some() automatically returnse false, but we want to return true.
+      if (project.pasPackages.length === 0) return true;
+      return project.pasPackages.some((projectPackage) => {
+        if (
+          projectPackage.statuscode === PACKAGE_STATUS_CODES.PACKAGE_PREPARATION
+          && [
+            PACKAGE_VISIBILITY_CODES.APPLICANT_ONLY,
+            PACKAGE_VISIBILITY_CODES.GENERAL_PUBLIC,
+          ].includes(projectPackage['dcp-visibility'])
+        ) {
+          return false;
+        }
         return true;
-      } return false;
+      });
     });
   }
+
+
+  @sort('applicantProjects', function(projectA, projectB) {
```

Comment:
As per PR review discussion, please add a "TODO: " note to flag a future enhancement on the sorting logic to account for instances when project titles start with a number (ex. 123 should come _before_ 8 -- which this alphabetical sorting doesn't do yet).